python/lsst/summit/utils/astrometry/fastStarTrackerAnalysis.py

```diff
@@ -0,0 +1,551 @@
+        toPrint.append(f'Mean and stddev of fluxes = {fluxMean:.1f} ± {fluxStd:.1f} ADU')
+
+    if not silent:
+        for line in toPrint:
+            print(line)
+
+    return consistent
+
+
+def plotResults(results, sourceIndex=0, allowInconsistent=False):
+    """Plot the centroid movements and fluxes etc for a set of results.
+
+    Parameters
+    ----------
+    results : `dict` of `list` of
+              `lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source`
+        A dict, keyed by sequence number, with each value being a list of the
+        sources found in the image, e.g. as returned by
+        ``analyzeFastStarTrackerImage()``.
+    sourceIndex : `int`, optional
+        If there is more than one source in every image, which source number
+        should the plot be made for? Defaults to zero, which is the brightest
+        source by default.
+    allowInconsistent : `bool`, optional
+        Make the plots even if the input results appear to be inconsistent?
+    """
+    consistent = checkResultConsistency(results.values(), silent=True)
+    if not consistent and not allowInconsistent:
+        checkResultConsistency(results.values(), silent=False)  # print the problem if we're raising
+        raise ValueError('The sources were found to be inconsistent and allowInconsistent=False')
+
+    sourceDict = {k: v[sourceIndex] for k, v in results.items()}
+    seqNums = list(sourceDict.keys())
+    sources = list(sourceDict.values())
+
+    axisLabelSize = 18
+
+    fig = plt.figure(figsize=(10, 16))
+    ax1, ax2, ax3 = fig.subplots(3, sharex=True)
+    fig.subplots_adjust(hspace=0)
+
+    ax1.plot(seqNums, [s.rawFlux for s in sources], label='Raw Flux')
+    ax1.plot(seqNums, [s.hsmFittedFlux for s in sources], label='Fitted Flux')
+    ax1.set_ylabel('Flux (ADU)', size=axisLabelSize)
+    ax1.legend()
+
+    ax2.plot(seqNums, [s.centroidX for s in sources], label='Raw centroid x')
+    ax2.plot(seqNums, [s.hsmCentroidX for s in sources], label='Fitted centroid x')
+    ax2.set_ylabel('x-centroid (pixels)', size=axisLabelSize)
+    ax2.legend()
+
+    ax3.plot(seqNums, [s.centroidY for s in sources], label='Raw centroid y')
+    ax3.plot(seqNums, [s.hsmCentroidY for s in sources], label='Fitted centroid y')
+    ax3.set_ylabel('y-centroid (pixels)', size=axisLabelSize)
+    ax3.set_xlabel('SeqNum', size=axisLabelSize)
+    ax3.legend()
+
+    fig = plt.figure(figsize=(10, 10))
+    ax4 = fig.subplots(1)
+
+    colors = np.arange(len(sources))
+    # gnuplot2 has a nice balance of nothing white, and having an intuitive
+    # progression of colours so the eye can pick out trends on the point cloud.
+    ax4.scatter([s.centroidX for s in sources], [s.centroidY for s in sources], c=colors, cmap='gnuplot2')
+    ax4.set_xlabel('x-centroid (pixels)', size=axisLabelSize)
+    ax4.set_ylabel('y-centroid (pixels)', size=axisLabelSize)
+    ax4.set_aspect('equal', 'box')
```

Comment:
Can you add a labelled colorbar or something else to indicate that color corresponds to the image?

Reply:
Hmmm, interesting. It's the position in the sequence, let me take a look at that.

Reply:
Done 🙂